lib/active_merchant/billing/gateways/midtrans.rb

```diff
@@ -98,10 +98,12 @@ def initialize(options={})
 
       def purchase(money, payment, options={})
         post = {}
+        @midtrans_gateway.config.override_notif_url = options[:notification_url] if options[:notification_url]

```

Comment:
I have not gone through the current implementation of `Midtrans` , but if possible can we set this in `initialize` method? 

If not, we can move this in a private function like how we did for `add_metadata`.